lib/index_documents.rb

```diff
@@ -0,0 +1,102 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  MIGRATED_TAGGING_APPS = %w{
+    businesssupportfinder
+    calculators
+    calendars
+    hmrc-manuals-api
+    licencefinder
+    policy-publisher
+    smartanswers
+  }
+
+  def process(message)
+    if publishing_app_migrated?(message)
+      raw_links = links_from_payload(message)
+
+      unless raw_links.empty?
+        links = rummager_fields_from_links(raw_links)
+        base_path = document_base_path(message)
+        update_index(base_path, links)
+      end
+    end
+
+    message.ack
+  end
+
+private
+  def publishing_app_migrated?(message)
+    MIGRATED_TAGGING_APPS.include? message.payload["publishing_app"]
+  end
+
+  def links_from_payload(message)
+    message.payload.fetch("links", {})
+  end
+
+  def rummager_fields_from_links(links)
+    results = {}
+    rummager_field_mappers.each { |field_name, mapper|
+      field_values = mapper.call(links)
+      if field_values
+        results[field_name] = field_values
+      end
+    }
+    results
+  end
+
+  def rummager_field_mappers
+    mapper = {}
+
+    mapper["mainstream_browse_pages"] = lambda { |links| sorted_link_paths(links, "mainstream_browse_page") }
```

Comment:
The field in the links hash is `mainstream_browse_pages` - https://github.com/alphagov/govuk-content-schemas/blob/master/formats/base_links.json#L10